cmd/cainjector/app/start.go

```diff
@@ -117,15 +117,16 @@ servers and webhook servers.`,
 
 func (o InjectorControllerOptions) RunInjectorController(ctx context.Context) error {
 	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
-		Scheme:                  api.Scheme,
-		Namespace:               o.Namespace,
-		LeaderElection:          o.LeaderElect,
-		LeaderElectionNamespace: o.LeaderElectionNamespace,
-		LeaderElectionID:        "cert-manager-cainjector-leader-election",
-		LeaseDuration:           &o.LeaseDuration,
-		RenewDeadline:           &o.RenewDeadline,
-		RetryPeriod:             &o.RetryPeriod,
-		MetricsBindAddress:      "0",
+		Scheme:                        api.Scheme,
+		Namespace:                     o.Namespace,
+		LeaderElection:                o.LeaderElect,
+		LeaderElectionNamespace:       o.LeaderElectionNamespace,
+		LeaderElectionID:              "cert-manager-cainjector-leader-election",
+		LeaderElectionReleaseOnCancel: true,
```

Comment:
The context cancelation will also shut down all other go routines. So the only situation where this could be a problem is if 1) the goroutine does not end when the context is canceled (this is a bug that should be fixed) or 2) the cleanup procedure (after the context cancelation) performs api-requests that can interfere with a newly started controller.

I don't think that situation 1 or situation 2 occurs in the codebase.